### PR TITLE
#444 - Credentials.FromStorageYaml fully implemented

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #444 
I've fully implemented `Credentials.FromStorageYaml` and added puzzles for `DeleteUserSlice` and for `AddUpdateUserSlice`.

### CODE REVIEW

File: src/main/java/com/artipie/Credentials.java [hunk 1]
```diff
@@ -89,19 +99,42 @@ public FromStorageYaml(final Storage storage, final Key key) {
...
+                            .add("pass", String.format("plain:%s", pswd))
+                            .build()
+                    );
+                    return FromStorageYaml.buildCredentials(result);
+                }
+            ).thenCompose(
+                updated -> this.storage.save(
+                    this.key, new Content.From(updated.getBytes(StandardCharsets.UTF_8))
+                )
+            );
```
Comment (olegmoz) [inline src/main/java/com/artipie/Credentials.java:124]: @olenagerasimova this code snippet is repeated in `remove` method and it also proceeded with `FromStorageYaml.buildCredentials`. I'd suggest to extract method `save(YamlMapping)` that will build credentials YAML and save it to storage

File: src/test/java/com/artipie/CredentialsFromStorageYamlTest.java [hunk 2]
```diff
@@ -23,38 +23,145 @@
...
+            .toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            new PublisherAs(this.storage.value(this.key).join())
+                .asciiString().toCompletableFuture().join(),
+            new IsEqual<>(this.getYaml(new ImmutablePair<>(mark, pass)))
+        );
+    }
+
+    @Test
+    void doNotChangeYamlIfUserNotFound() {
```
Comment (olegmoz) [inline src/test/java/com/artipie/CredentialsFromStorageYamlTest.java:129]: @olenagerasimova please specify in this test method name that we are talking about removing user

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected, take a look please

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#478` is not assigned, can't get performer

**Pull request merged**